crates/wasi/src/view.rs

```diff
@@ -0,0 +1,86 @@
+use crate::ctx::WasiCtx;
+use wasmtime::component::ResourceTable;
+
+pub trait IoView: Send {
```

Comment:
To perhaps bikeshed a bit, maybe `TableView`?

Reply:
Also mind ensuring there's documentation for this trait as well? (similar to the `WasiView` one below)

Reply:
I named it `Io` instead of `Table` view in anticipation of factoring it out for the `wasmtime-wasi-io` crate. I don't mind renaming it, but thats why I was thinking `Io` over `Table`.